code/modules/mapping/mapping_helpers.dm

```diff
@@ -328,31 +328,100 @@ CREATION_TEST_IGNORE_SUBTYPES(/obj/effect/mapping_helpers)
 		log_mapping("[src] failed to find air alarm at [AREACOORD(src)].")
 	qdel(src)
 
-//APC helpers
 /obj/effect/mapping_helpers/apc
+	name = "apc helpers"
+	icon_state = "apc_helper"
 
 /obj/effect/mapping_helpers/apc/Initialize(mapload)
 	. = ..()
 	if(!mapload)
 		log_mapping("[src] spawned outside of mapload!")
-		return
-	var/obj/machinery/power/apc/apc = locate(/obj/machinery/power/apc) in loc
-	if(!apc)
-		log_mapping("[src] failed to find an APC at [AREACOORD(src)]")
+		return INITIALIZE_HINT_QDEL
+	var/obj/machinery/power/apc/target = locate(/obj/machinery/power/apc) in loc
+	if(isnull(target))
+		var/area/target_area = get_area(target)
+		log_mapping("[src] failed to find an apc at [AREACOORD(src)] ([target_area.type]).")
 	else
-		payload(apc)
+		payload(target)
+
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/effect/mapping_helpers/apc/LateInitialize()
+	. = ..()
+	var/obj/machinery/power/apc/target = locate(/obj/machinery/power/apc) in loc
+
+	if(isnull(target))
+		qdel(src)
+		return
+	if(target.unlocked)
+		target.unlock()
+	if(target.syndicate_access)
+		target.give_syndicate_access()
+	if(target.away_engine_access)
+		target.give_away_engine_access()
+	if(target.exploration_access)
+		target.give_exploration_access()
+
+	target.update_icon()
+	qdel(src)
 
 /obj/effect/mapping_helpers/apc/proc/payload(obj/machinery/power/apc/payload)
 	return
 
-/obj/effect/mapping_helpers/apc/discharged
-	name = "apc zero change helper"
+/obj/effect/mapping_helpers/apc/charge
+	name = "apc change helper"
 	icon_state = "apc_nopower"
+	var/charge_percentage = 100
```

Comment:
most people are probably going to want to default to 0 instead of the default charge level.